azure-functions/src/durable/history.rs

```diff
@@ -0,0 +1,80 @@
+use chrono::{DateTime, FixedOffset};
+use serde::Deserialize;
+use serde_repr::Deserialize_repr;
+
+// TODO refactor this to make enum HistoryEvent that for each value it will have its own struct
+// enum HistoryEvent { ExecutionStarted(ExecutionStartedEvent), ... }
+// serde now doesn't support elegant conversion from json to tagged enums with custom tag value out of the box.
+// i.e conversion { EventType = 0, EventId = 1, ...} => ExecutionStarted(ExecutionStartedEvent)
+// in future we can implement manual translation
+
+#[doc(hidden)]
+#[derive(Debug, Clone, Deserialize, PartialEq)]
+#[serde(rename_all = "PascalCase")]
+pub struct HistoryEvent {
```

Comment:
Does this need to be `pub`?  I don't think we'll be exposing this data on the `DurableOrchestrationContext`, only using it internally.

Reply:
I used this type in `DurableOrchestartionContext` tests. So it needs some kind of visibility outside current mod.

Reply:
Would `pub(crate)` work?

Reply:
Yeah, that's what I did just now